neural_lam/weather_dataset.py

```diff
@@ -126,7 +259,7 @@ def __len__(self):
                 warnings.warn(
                     "only using first ensemble member, so dataset size is "
                     " effectively reduced by the number of ensemble members "
-                    f"({self.da_state.ensemble_member.size})",
+                    f"({self.datastore._num_ensemble_members})",
```

Comment:
While we do not handle the case of ensemble reanalysis data atm, it might be nice to allow for extending to that in the future. Therefore I would probably prefer to keep `is_ensemble` and `is_forecast` as orthogonal properties. And not do `is_forecast = (num_ensemble_members > 0)` (if that is how you propose to do that derivation?).

Reply:
I anyhow think that better handling of `is_forecast`/`is_ensemble` is orthogonal to this PR, and is better addressed separately to prevent scope creep of this change.

Reply:
I have implemented this in BaseDatastore.num_ensemble_members
Having  `BaseDatastore.num_ensemble_members` allows to treat `is_ensemble` and `is_forecast` as 2 truly independent attributes.

Reply:
Haha Joel I think we had the same ideas there. Hope my solution is a good compromise that allows for future mdp-ensembles, while keeping this PR as simple as possible.

Reply:
Haha Joel I think we had the same ideas there. Hope my solution is a good compromise that allows for future mdp-ensembles, while keeping this PR as simple as possible.

Reply:
I forgot to declare the new abstract method in the mdp datastore: num_ensemble_members

Reply:
I see, we were writing at the same time :smile: good! Would it make sense to already in BaseDatastore default `num_ensemble_members` to 1 (not an ensemble) and derive `is_ensemble` from `num_ensemble_members > 1`? One could then override `num_ensemble_members` in subclasses that allow for working with ensemble data.

Reply:
Yep, that makes more sense, implemented in: c0c50d5e44580c6a8e486d44cee19be6ed1fd847